test/integ/basic.cpp

```diff
@@ -304,11 +308,21 @@ class print_response_resource : public http_resource {
      stringstream* ss;
 };
 
+#ifdef HTTPSERVER_PORT
+#define PORT HTTPSERVER_PORT
+#else
+#define PORT 8080
+#endif  // PORT
+
+#define STR2(p) #p
+#define STR(p) STR2(p)
```

Comment:
Why do we need the proxy definition of STR(p)? Cannot we just simply have what's on line 317 and name that STR(p) ?

Reply:
Without the extra layer, the macro expands to the literal argument (`PORT`) rather than what that argument expands to:

A quick demo of how things expand: https://godbolt.org/z/Mz4eexzb5

See also: https://gcc.gnu.org/onlinedocs/gcc-4.8.5/cpp/Stringification.html

Reply:
Makes sense, thanks.